### PR TITLE
[ci] Moved windows build & test to github action

### PR DESCRIPTION
**Pull request by Leonz5288**
Related issue = #2428

### CODE REVIEW

File: python/taichi/diagnose.py [hunk 1]
```diff
@@ -39,7 +39,10 @@ def try_print(tag, expr):
         try:
             cmd = f'import taichi as ti; print("===="); print({expr}, end="")'
             ret = subprocess.check_output([executable, '-c', cmd]).decode()
-            ret = ret.split('====\n', maxsplit=1)[1]
+            if platform.system() == "Windows":
```
Comment (k-ye) [inline python/taichi/diagnose.py:42]: Let's use single quote '

File: tests/python/test_indices_assert.py [hunk 2]
```diff
@@ -1,11 +1,16 @@
-import taichi as ti
+import platform
+
 import pytest
 
+import taichi as ti
+
 
+@pytest.mark.skipif(platform.system() == "Windows",
```
Comment (k-ye) [inline tests/python/test_indices_assert.py:8]: ditto

File: python/taichi/diagnose.py [hunk 3]
```diff
@@ -39,7 +39,10 @@ def try_print(tag, expr):
...
+                ret = ret.split('===\r\n', maxsplit=1)[1]
```
Comment (k-ye) [inline python/taichi/diagnose.py:43]: better use `os.linesep`?

**Review by k-ye — COMMENTED** [review]
Great work!

**Comment by Leonz5288** [comment]
I think we are ready to delete Appveyor and merge @k-ye

**Review by k-ye — APPROVED** [review]
LGTM!

> I think we are ready to delete Appveyor and merge

We still need to include this to the release process before fully deprecating appveyor

**Pull request merged**